muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/user/rest/UserEntityFileRESTService.java

```diff
@@ -164,6 +165,44 @@ else if (!userEntityFile.getUserEntity().getId().equals(loggedUserEntity.getId()
       .type(userEntityFile.getContentType())
       .build();
   }
+  
+  @DELETE
+  @Path("/{USERENTITYID}/identifier/{IDENTIFIER}")
+  @RESTPermit (handling = Handling.INLINE)
+  public Response deleteFile(@PathParam("USERENTITYID") Long userEntityId, @PathParam("IDENTIFIER") String identifier, @Context Request request) {
+	// Check if the file exist
+	UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+	if (userEntity == null) {
+	  return Response.status(Status.NOT_FOUND).build();
+	}
+	UserEntityFile userEntityFile = userEntityFileController.findByUserEntityAndIdentifier(userEntity, identifier);
+	if (userEntityFile == null) {
+	  return Response.status(Status.NOT_FOUND).build();
+	}
+	
+	//Check wheter the logged in user is the same as this user or whether it's an administrator
+	UserEntity loggedUserEntity = sessionController.getLoggedUserEntity();
+	if (loggedUserEntity == null) {
+		return Response.status(Status.FORBIDDEN).build();
+	}
+	EnvironmentUser environmentUser = environmentUserController.findEnvironmentUserByUserEntity(loggedUserEntity);
+	boolean isOwnerOfTheFileAndFileIsPublic = userEntity.getId().equals(loggedUserEntity.getId()) && userEntityFile.getVisibility() == UserEntityFileVisibility.PUBLIC;
```

Comment:
This will prevent users from deleting their own files unless they are PUBLIC. Should be allowed if:

- you are the owner of the file, whatever its visibility
- you are a staff member and the visibility of the file is either PUBLIC or STAFF

Also, a more compact way to determine staff member would be simply to see if EnvironmentRoleArchetype is not STUDENT.